transport-classes-epoll/src/main/java/io/netty/channel/epoll/AbstractEpollStreamChannel.java

```diff
@@ -1043,6 +1043,12 @@ public boolean spliceIn(RecvByteBufAllocator.Handle handle) {
     private final class EpollSocketWritableByteChannel extends SocketWritableByteChannel {
         EpollSocketWritableByteChannel() {
             super(socket);
+            assert fd == socket;
+        }
+
+        @Override
+        protected int write(final ByteBuffer buf, final int pos, final int limit) throws IOException {
+            return socket.send(buf, pos, limit);
```

Comment:
That's an hack because `SocketWritableByteChannel' is a public API and I cannot introduce a generic param `T extends FileDescriptor` on it @normanmaurer

Reply:
I really don't like this trick @normanmaurer :( I feel dirty